src/Spain/Mews.Fiscalizations.Spain/Communication/SoapClient.cs

```diff
@@ -28,22 +30,33 @@ internal SoapClient(Uri endpointUri, X509Certificate certificate, TimeSpan httpT
 
         private HttpClient HttpClient { get; }
 
-        internal async Task<TOut> SendAsync<TIn, TOut>(TIn messageBodyObject)
+        internal async Task<ResponseResult<TOut>> SendAsync<TIn, TOut>(TIn messageBodyObject)
             where TIn : class, new()
             where TOut : class, new()
         {
             var parameters = new XmlSerializationParameters(namespaces: GetSiiNameSpaces());
             var messageBodyXmlElement = XmlSerializer.Serialize(messageBodyObject, parameters);
             XmlMessageSerialized?.Invoke(this, new XmlMessageSerializedEventArgs(messageBodyXmlElement));
 
-            var soapMessage = new SoapMessage(messageBodyXmlElement);
-            var xmlDocument = soapMessage.GetXmlDocument();
+            var soapRequestMessage = new SoapMessage(messageBodyXmlElement);
+            var xmlDocument = soapRequestMessage.GetXmlDocument();
             var xml = xmlDocument.OuterXml;
 
             var response = await GetResponseAsync(xml);
-
-            var soapBody = GetSoapBody(response);
-            return XmlSerializer.Deserialize<TOut>(soapBody.OuterXml);
+            try
+            {
+                var soapResponseBody = GetSoapBody(response);
+                var deserializedMessage = XmlSerializer.Deserialize<TOut>(soapResponseBody.OuterXml);
+                return new ResponseResult<TOut>(successResult: deserializedMessage);
+            }
+            catch (InvalidOperationException)
+            {
+                var soapFault = XmlSerializer.Deserialize<SoapFaultResponse>(response);
+                return new ResponseResult<TOut>(errorResult: new ErrorResult(
+                    code: soapFault.Body.Fault.Faultcode,
+                    message: soapFault.Body.Fault.Faultstring
+                ));
+            }
```

Comment:
Unfortunately, the SII returns a success status code 200 when there is an issue with the XML (SOAPFault)

The same thing for validation errors, they are returned as a part of the invoice response.